src/gui/tray/UserModel.cpp

```diff
@@ -435,56 +435,62 @@ void User::slotItemCompleted(const QString &folder, const SyncFileItemPtr &item)
     if (!folderInstance)
         return;
 
-    if (isValueableActivity(folderInstance, item)) {
-        qCWarning(lcActivity) << "Item " << item->_file << " retrieved resulted in " << item->_errorString;
+    if (!isActivityOfCurrentAccount(folderInstance)) {
+        return;
+    }
 
-        Activity activity;
-        activity._type = Activity::SyncFileItemType; //client activity
-        activity._status = item->_status;
-        activity._dateTime = QDateTime::currentDateTime();
-        activity._message = item->_originalFile;
-        activity._link = folderInstance->accountState()->account()->url();
-        activity._accName = folderInstance->accountState()->account()->displayName();
-        activity._file = item->_file;
-        activity._folder = folder;
-        activity._fileAction = "";
-
-        if (item->_instruction == CSYNC_INSTRUCTION_REMOVE) {
-            activity._fileAction = "file_deleted";
-        } else if (item->_instruction == CSYNC_INSTRUCTION_NEW) {
-            activity._fileAction = "file_created";
-        } else if (item->_instruction == CSYNC_INSTRUCTION_RENAME) {
-            activity._fileAction = "file_renamed";
+    if (isActivityIgnored(item)) {
```

Comment:
Nope. :-)

What I meant is that there's a business rule, I can't see it at the call site. Beside with such a name why isn't it containing the current account check? (and then we'd be back where we started from)

It needs to say what it does: `isUnsolvableConflict`.

Which is why I proposed to use `const auto` variables in my previous comment. At that point the functions don't improve the homogeneity in level of abstraction of that code (which would be primary motive for functions) because of the rest of the method and they're called only once. They just add syntactic noise and scaffolding.

Still I'm willing to roll with them as long as they are properly named. They're private and called only once, they can change anytime.

Reply:
Okay I think I understand that. My initial idea was that at some point there might be more cases when we want to ignore activities. Therefore I thought it would be a good idea to group all this checks in a method and then just call this method once. But for now we just have this one case.

Reply:
> Emphasis above mine... and... YAGNI ;-)
True;)